elytron/WFLY-16532-additional-scope-for-auth-request.adoc

```diff
@@ -0,0 +1,131 @@
+== Adding the ability to configure additional scope value for an authentication request
+:author:            Prarthona Paul
+:email:             prpaul@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+OpenID Connect is an authentication mechanism that builds on OAuth 2.0 
+and allows a user to login to a web application using credentials established 
+by an OpenID provider. 
+Currently, when sending an authentication request to the OpenID provider, one 
+of the required parameters with the authentication code flow is "scope". However, for 
+now, the Elytron OIDC HTTP authentication mechanism hardcodes this value to just "openid".
+
+The https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest[OpenID Connect specification] indicate that there are other scope values which may be included in 
+the authentication request. This new feature will add the ability to configure the `scope` attribute
+of the `elytron-oidc-client` subsystem under the `secure-server` and `secure-deployment` resources, so that additional scope values can be specified when 
+configuring the server or the deployment settings. 
+
+The feature will allow the user to specify additional scope values in two ways: 
+
+* In an application's `oidc.json` configuration file in the `WEB-INF` directory of the application,
+
+* Adding configurations to the `elytron-oidc-client` subsystem under the `secure-deployment` and the 'secure-server' resource.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-16532[WFLY-16532]
+
+* https://issues.redhat.com/browse/ELY-2574[ELY-2574]
+
+
+=== Related Issues
+
+* N/A
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* WildFly
+
+* WildFly Elytron
+
+=== Other Interested Projects
+
+N/A
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* A new attribute named `scope` will be added to the `secure-deployment` resource under the `elytron-oidc-client` subsystem, which will be used 
+to specify additional scope values. These values will be used by the Elytron HTTP OIDC authentication mechanism.  
+
+* It must be possible to configure this attribute using cli commands. For example: 
+
+```
+    /subsystem=elytron-oidc-client/secure-deployment=my-secure-deployment:write-attribute(name=scope, value="openid offline_access")
```

Comment:
I think for the examples here and below, we can remove "openid". We can mention that this will automatically be added when forming the authentication request if not present.

Reply:
fixed!